BigQuery/src/Model.php

```diff
@@ -0,0 +1,229 @@
+<?php
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery;
+
+use Google\Cloud\BigQuery\Connection\ConnectionInterface;
+use Google\Cloud\Core\ConcurrencyControlTrait;
+use Google\Cloud\Core\Exception\NotFoundException;
+
+/**
+ * [Models](https://cloud.google.com/bigquery-ml/docs/) in BigQuery ML represent what an ML system
+ * has learned from training the data.
+ *
+ * @package Google\Cloud\BigQuery
+ */
+class Model
+{
+    use ConcurrencyControlTrait;
+
+    /**
+     * @var ConnectionInterface Represents a connection to BigQuery.
+     */
+    private $connection;
+
+    /**
+     * @var array The model's identity
+     */
+    private $identity = [];
+
+    /**
+     * @var array The model's metadata.
+     */
+    private $info = [];
+
+    /**
+     * @param ConnectionInterface $connection Represents a connection to BigQuery.
+     * @param $id string The model's ID.
+     * @param $datasetId string The dataset's ID.
+     * @param $projectId string The project's ID.
+     * @param $info array The model's details
+     */
+    public function __construct(
+        ConnectionInterface $connection,
+        $id,
+        $datasetId,
+        $projectId,
+        array $info = []
+    ) {
+        $this->connection = $connection;
+        $this->identity = [
+            'modelId' => $id,
+            'datasetId' => $datasetId,
+            'projectId' => $projectId
+        ];
+
+        $this->info = $info;
+    }
+
+    /**
+     * Retrieve's the model's details. If no model data is cached, a network request
+     * will be made to retrieve it.
+     *
+     * Example:
+     * ```
+     * $info = $model->info();
+     * echo $info['modelType'];
+     * ```
+     *
+     * @see https://cloud.google.com/bigquery/docs/reference/rest/v2/models#resource Model resource documentation.
+     * @param array $options
+     * @return array
+     */
+    public function info(array $options = [])
+    {
+        if (!$this->info) {
```

Comment:
nit: can we write this with a shorthand ternary?

```php
return $this->info ?: $this->reload($options);
```

Reply:
Sure can!